analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/AnalysisScaffolding.cs

```diff
@@ -80,6 +81,33 @@ public static string CreateSonarProjectConfigWithFilesToAnalyze(TestContext cont
         public static string CreateSonarProjectConfig(TestContext context, ProjectType projectType, bool isScannerRun = true) =>
             CreateSonarProjectConfig(context, "ProjectType", projectType.ToString(), isScannerRun);
 
+        public static string CreateSonarLintXml(
+            TestContext context,
+            bool ignoreHeaderComments = false,
+            bool analyzeGeneratedCode = false,
+            string[] exclusions = null,
+            string[] inclusions = null,
+            string[] globalExclusions = null,
+            string[] testExclusions = null,
+            string[] testInclusions = null,
+            string[] globalTestExclusions = null) =>
+            TestHelper.WriteFile(context, "SonarLint.xml",
+                new XDocument(
+                    new XDeclaration("1.0", "utf-8", "yes"),
+                    new XElement("AnalysisInput",
+                        new XElement("Settings",
+                            CreateKeyValuePair("Setting", "sonar.cs.ignoreHeaderComments", ignoreHeaderComments.ToString()),
```

Comment:
I suggest providing the language to the method and adding also UTs that check the functionality works for Vb.Net

Reply:
What kind of functionality? The language doesn't change much within the exclusions/inclusions context since they don't have the language (`cs`/`vbnet`) inside the property name (es: `sonar.cs.analyzeGeneratedCode`).

Reply:
Actually sorry, after taking another look I realized you don't use this method to test the header comments and analyzeGeneratedCode parameters at all, so you can rename the method to reflect that it sets only the exclusion inclusion parameters, and only write those to the file.

Reply:
I added those properties to have a more generic method that can be used to add more test cases for those properties as well. Do you think it's better to reduce it now and add the properties later if needed?

Reply:
Wait, there is a CreateSonarLintXml method on SonarAnalysisContextBaseTest.ShouldAnalyzeTree to test the analyzeGeneratedProperty that creates a hardcoded SonarLint.xml file. I can replace that logic with this builder, this way we will need those properties and we reduce duplication. WDYT?

Reply:
You can do it, and at the same time the ingoreHeader parameter will still be unused, so I would remove it. I suggest keeping things simple.